core/src/main/kotlin/entity/interaction/Interaction.kt

```diff
@@ -41,27 +41,75 @@ class Interaction(
 
     override val id: Snowflake get() = data.id
 
-    override val channelId: Snowflake get() = data.channelId
-
     override val token: String get() = data.token
 
-    override val guildId: Snowflake get() = data.guildId
-
     val type: InteractionType get() = data.type
 
-    val permissions: Permissions get() = data.permissions
-
-    val channel: TextChannelBehavior get() = TextChannelBehavior(id = channelId, guildId = guildId, kord = kord)
-
-    val guild get() = GuildBehavior(guildId, kord)
-
-    val member: MemberBehavior get() = MemberBehavior(data.guildId, data.member.userId, kord)
+    val permissions: Permissions? get() = data.permissions
 
     val command: Command
         get() = Command(data.data)
 
     val version: Int get() = data.version
 
+    companion object {
+        operator fun invoke(data: InteractionData, applicationId: Snowflake, kord: Kord): Interaction {
+            return when {
+                data.user != null -> DMInteraction(data, applicationId, kord)
+                data.channelId != null && data.guildId != null && data.member != null ->
```

Comment:
I do believe channelIds are available in both cases cause a DM Channel has a id as well

Reply:
```json
{
  "channel_id": "637662290170216458",
  "data": {
    "id": "812857383717699626",
    "name": "online"
  },
  "id": "814909846746431539",
  "token": "-",
  "type": 2,
  "user": {
    "avatar": "8bd2b747f135c65fd2da873c34ba485c",
    "discriminator": "4185",
    "id": "123170274651668480",
    "public_flags": 131712,
    "username": "MrPowerGamerBR"
  },
  "version": 1
}
```
confirmed with this sent by @MrPowerGamerBR

Reply:
Also, `channel_id` is nullable if you are looking at the docs: https://discord.com/developers/docs/interactions/slash-commands#interaction

*But* Mason says that it is a "future proof" thing and that for every use case that exists, you will receive a Channel ID: https://github.com/discord/discord-api-docs/pull/2568/files#r569025697